lib/fs.js

```diff
@@ -2008,13 +2008,13 @@ function ReadStream(path, options) {
   this.closed = false;
 
   if (this.start !== undefined) {
-    if (typeof this.start !== 'number') {
-      throw new ERR_INVALID_ARG_TYPE('start', 'number', this.start);
+    if (typeof this.start !== 'number' || !Number.isInteger(this.start)) {
```

Comment:
Nit: the condition can be just `!Number.isInteger()` as that returns `true` when passing something that is not a number.

Reply:
Yes: http://www.ecma-international.org/ecma-262/8.0/index.html#sec-number.isinteger